steppy/base.py

```diff
@@ -1,131 +1,132 @@
+import glob
 import os
 import pprint
 import shutil
 from collections import defaultdict
 
 from sklearn.externals import joblib
 
-from steppy.adapter import AdapterError
-from steppy.utils import view_graph, save_graph, get_logger, initialize_logger
+from steppy.adapter import Adapter, AdapterError
+from steppy.utils import display_pipeline, persist_as_png, get_logger, initialize_logger
 
 initialize_logger()
 logger = get_logger()
 
 
 class Step:
-    """Building block of steps pipelines.
-
-    Step is an execution wrapper over the transformer (see BaseTransformer class documentation) that enables building complex machine learning pipelines.
-    It handles multiple input/output data flows, has build-in persistence/caching of both models (transformers) and
-    intermediate step outputs.
-    Step executes fit_transform on every step recursively starting from the very last step and making its way forward
-    through the input_steps. If step transformer was fitted already then said transformer is loaded in and the transform method
-    is executed.
-    One can easily debug the data flow by plotting the pipeline graph with either step.save_graph(filepath) method
-    or simply returning it in a jupyter notebook cell.
-    Every part of the pipeline can be easily accessed via step.get_step(name) method which makes it easy to reuse parts of the pipeline
-    across multiple solutions.
+    """Building block of steppy pipelines.
+
+    Step is an execution wrapper over the transformer (see BaseTransformer), which realizes single operation on data. With Step you can:
+        1. design multiple input/output data flows and connections between Steps.
+        2. handle persistence and caching of transformers and intermediate results.
+    Step executes 'fit_transform' method inspired by the sklearn on every step recursively starting from the very last Step and making its way forward through the input_steps.
+    One can easily debug the data flow by plotting the pipeline graph (see: step.persist_as_png(filepath) or return step in a jupyter notebook cell.
     """
     def __init__(self,
                  name,
                  transformer,
-                 input_steps=None,
+                 experiment_directory,
```

Comment:
This is why I like to do `def fun(self, *, arg1, arg2, arg3)`. Without it a small change like adding another argument can mess up a lot in code that uses this functions/class.

Reply:
So you recommend sth like:
`     def __init__(self, *,  name, transformer, ...):`